src/http/request_template.rs

```diff
@@ -100,12 +99,32 @@ impl RequestTemplate {
     Ok(req)
   }
 
+
   /// Sets the body for the request
-  fn set_body<C: PathString + HasHeaders>(&self, mut req: reqwest::Request, ctx: &C) -> reqwest::Request {
+  pub fn set_body<C: PathString + HasHeaders>(&self, mut req: reqwest::Request, ctx: &C) -> reqwest::Request {
     if let Some(body) = &self.body {
-      req.body_mut().replace(body.render(ctx).into());
+        // Checks and sets content type based on your condition
+        if Self::is_application_x_www_form_urlencoded(&req.headers()) {
+            req.headers_mut().insert(CONTENT_TYPE, HeaderValue::from_static("application/x-www-form-urlencoded"));
+
+            // Serialize the Mustache template directly to form_urlencoded
+            let form_data = serde_urlencoded::to_string(&body.render(ctx)).unwrap();
+            req.body_mut().replace(form_data.into());
+        }
+
+        req.body_mut().replace(body.render(ctx).into());
     }
     req
+}
+
+  /// checks for the CONTENT_TYPE header
+  pub fn is_application_x_www_form_urlencoded(headers: &HeaderMap) -> bool {
+    if let Some(content_type) = headers.get(CONTENT_TYPE) {
+        if let Ok(content_type_str) = content_type.to_str() {
+            return content_type_str.to_lowercase().starts_with("application/x-www-form-urlencoded");
+        }
+    }
+    false
```

Comment:
We could have the headers parsed into a Rust enum so we don't have something specific to `form-urlencoded`. Instead, we receive an enum and handle explicitly each case.

Two approaches can be taken:
- Have one variation to each implemented enum, plus a non-implemented variation
  - e.g., if matches `application/x-www-form-urlencoded` is urlencoded, else if matches `application/json`, otherwise it'll be the non-implemented variation.
- Consider `application/json` the default and add more variations to each new implemented
  - e.g., if matches `application/x-www-form-urlencoded` is urlencoded, otherwise it'll always be json

Reply:
@ologbonowiwi Thanks for the help.🙂